### PR TITLE
rename: add new package

### PR DESCRIPTION
**Pull request by davydden**

### CODE REVIEW

**Comment by citibeth** [comment]
Why does this have no dependencies?  Shouldn't it depend on `perl`?

**Comment by davydden** [comment]
@tgamblin @citibeth dependency added, i guess it should be `nolink`.

File: var/spack/repos/builtin/packages/rename/package.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+class Rename(Package):
+    """Perl-powered file rename script with many helpful built-ins."""
+
+    homepage = "http://plasmasturm.org/code/rename"
+    url      = "https://github.com/ap/rename/archive/v1.600.tar.gz"
+
+    version('1.600', '91beb555c93d407420b5dad191069bb3')
+
+    depends_on('perl', type='nolink')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/rename/package.py]: Shouldn't it be `type='run'`? It doesn't seem like Perl is needed at build time.

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/rename/package.py]: Oh, maybe. Is that something that is present on all systems? Or should we add a `pod2man` package to Spack?

Comment (davydden) [inline var/spack/repos/builtin/packages/rename/package.py]: frankly, i have no idea. I would let it be until somebody encounter a problem.

**Comment by davydden** [comment]
@tgamblin ping.

File: var/spack/repos/builtin/packages/rename/package.py (hunk 1 above)
Comment (tgamblin) [inline var/spack/repos/builtin/packages/rename/package.py]: OT: How do people feel about getting rid of `nolink`, `norun`, and `nobuild`?  I am starting to wish we'd just gone with explicit tuples for these in the first place.  Also, right now we've only got three deptypes, but things will get less understandable if we add another.

Comment (davydden) [inline var/spack/repos/builtin/packages/rename/package.py]: > How do people feel about getting rid of nolink, norun, and nobuild?

I am in favour.

**Comment by citibeth** [comment]
Me too

File: var/spack/repos/builtin/packages/rename/package.py (hunk 1 above)
Comment (hartzell) [inline var/spack/repos/builtin/packages/rename/package.py]: `pod2man` is part of the Perl core.  If you have `perl`, you should have `pod2man`.  Unless, perhaps you installed your `perl` from some vendor's packages that disassembled things and package various bits separately.

The Spack perl packages install it:
```
[hartzelg@spack-0 rss]$ find spack-test-drive-2525/ | grep pod2man
spack-test-drive-2525/opt/spack/linux-centos7-x86_64/gcc-5.4.0/perl-5.24.0-xjz2cnt45rbcfpawhknn6lhd5wuuz3ra/bin/pod2man
spack-test-drive-2525/opt/spack/linux-centos7-x86_64/gcc-5.4.0/perl-5.24.0-xjz2cnt45rbcfpawhknn6lhd5wuuz3ra/man/man1/pod2man.1
spack-test-drive-2525/opt/spack/linux-centos7-x86_64/gcc-5.4.0/perl-5.22.2-secuh2wuhq3a3huxgnuplmfqc225pcht/bin/pod2man
spack-test-drive-2525/opt/spack/linux-centos7-x86_64/gcc-5.4.0/perl-5.22.2-secuh2wuhq3a3huxgnuplmfqc225pcht/man/man1/pod2man.1
```

**Pull request merged**